Simplenote/NoticePresenter.swift

```diff
@@ -0,0 +1,168 @@
+import UIKit
+
+class NoticePresenter {
+
+    // MARK: Properties
+    //
+    var noticeView: NoticeView?
+    private var containerView: PassthruView?
+    private var noticeVariableConstraint: NSLayoutConstraint?
+
+    private var keyboardHeight: CGFloat = .zero
+    private var keyboardFloats: Bool = false
+    private var keyboardNotificationTokens: [Any]?
+
+    private var keyWindow: UIWindow? {
+        return UIApplication.shared.windows.first(where: { $0.isKeyWindow })
+    }
+    private var windowFrame: CGRect {
+        return keyWindow?.frame ?? .zero
+    }
+
+    private var bottomConstraintConstant: CGFloat {
+        let constant = Constants.bottomMarginConstant
+
+        if keyboardFloats || keyboardHeight == .zero {
+            return constant
+        }
+
+        return constant - keyboardHeight
+    }
+
+    // MARK: Lifecycle
+    //
+    deinit {
+        stopListeningToKeyboardNotifications()
+    }
+
+    // MARK: Presenting Methods
+    //
+    func presentNoticeView(_ noticeView: NoticeView, completion: @escaping (Bool) -> Void) {
```

Comment:
I think having `Bool` in completion here is confusing :-) If it's false, does it mean it wasn't presented?

Reply:
I had done that so that I could use the same completion handler signature as UIView.animate... but I agree it is confusing, the bool isn't really needed there.  I removed it